template/runtime/src/precompiles.rs

```diff
@@ -5,31 +5,23 @@ use sp_std::marker::PhantomData;
 use pallet_evm_precompile_modexp::Modexp;
 use pallet_evm_precompile_sha3fips::Sha3FIPS256;
 use pallet_evm_precompile_simple::{ECRecover, ECRecoverPublicKey, Identity, Ripemd160, Sha256};
+use pallet_evm_precompile_xvm::XvmPrecompile;
 
 pub struct FrontierPrecompiles<R>(PhantomData<R>);
 
-impl<R> FrontierPrecompiles<R>
-where
-	R: pallet_evm::Config,
-{
+impl<R> FrontierPrecompiles<R> {
 	pub fn new() -> Self {
 		Self(Default::default())
 	}
-	pub fn used_addresses() -> [H160; 7] {
-		[
-			hash(1),
-			hash(2),
-			hash(3),
-			hash(4),
-			hash(5),
-			hash(1024),
-			hash(1025),
-		]
+	pub fn used_addresses() -> impl Iterator<Item = H160> {
+		[1, 2, 3, 4, 5, 1024, 1025, 20485].into_iter().map(hash)
```

Comment:
Maybe comment what these precompiles are?

Reply:
We should also try for an ed25519 precompile eventually so we can leverage this: https://github.com/ChorusOne/tendermint-sol

Reply:
https://github.com/GoldenGateGGX/golden-gate/issues/12